lib/ansible/modules/cloud/amazon/ec2_lc.py

```diff
@@ -225,125 +325,172 @@ def create_launch_config(connection, module):
+                                 "no launch configuration returned from boto3" +
+                                 " after create.")
+        except botocore.exceptions.ClientError as e:
             module.fail_json(msg=str(e))
 
-    result = dict(
-                 ((a[0], a[1]) for a in vars(launch_configs[0]).items()
-                  if a[0] not in ('connection', 'created_time', 'instance_monitoring', 'block_device_mappings'))
-        )
-    result['created_time'] = str(launch_configs[0].created_time)
-    # Looking at boto's launchconfig.py, it looks like this could be a boolean
-    # value or an object with an enabled attribute.  The enabled attribute
-    # could be a boolean or a string representation of a boolean.  Since
-    # I can't test all permutations myself to see if my reading of the code is
-    # correct, have to code this *very* defensively
-    if launch_configs[0].instance_monitoring is True:
-        result['instance_monitoring'] = True
+    result = (dict((k, v) for k, v in launch_config.items()
+              if k not in ['connection', 'created_time', 'instance_monitoring',
+              'block_device_mappings']))
+
+    result['CreatedTime'] = str(launch_config.get('CreatedTime'))
+
+    if launch_config.get('InstanceMonitoring') is True:
+        result['InstanceMonitoring'] = True
     else:
         try:
-            result['instance_monitoring'] = module.boolean(launch_configs[0].instance_monitoring.enabled)
+            result['InstanceMonitoring'] = module.boolean(launch_config.get(
+                'InstanceMonitoring').get('Enabled'))
         except AttributeError:
-            result['instance_monitoring'] = False
-    if launch_configs[0].block_device_mappings is not None:
-        result['block_device_mappings'] = []
-        for bdm in launch_configs[0].block_device_mappings:
-            result['block_device_mappings'].append(dict(device_name=bdm.device_name, virtual_name=bdm.virtual_name))
-            if bdm.ebs is not None:
-                result['block_device_mappings'][-1]['ebs'] = dict(snapshot_id=bdm.ebs.snapshot_id, volume_size=bdm.ebs.volume_size)
+            result['InstanceMonitoring'] = False
+    if launch_config.get('BlockDeviceMappings') is not None:
+        result['BlockDeviceMappings'] = []
+        for bdm in launch_config.get('BlockDeviceMappings'):
+            result['BlockDeviceMappings'].append(dict(device_name=bdm.get(
+                'DeviceName'), virtual_name=bdm.get('VirtualName')))
+            if bdm.get('Ebs') is not None:
+                result['BlockDeviceMappings'][-1]['ebs'] = dict(
+                    snapshot_id=bdm.get('Ebs').get('SnapshotId'),
+                    volume_size=bdm.get('Ebs').get('VolumeSize'))
 
     if user_data_path:
-        result['user_data'] = "hidden" # Otherwise, we dump binary to the user's terminal
+        result['UserData'] = "hidden"
+
+    return_object = {
```

Comment:
Can you document the return values please? This module is stableinterface so I could see that you may just be maintaining the current return structure but it isn't readily apparent.